### DISCUSSION TITLE
navigate to grid from another page using url

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6854] [Help] [answered]
I have sales amount displayed on the home page like a dashboard, when you click on it it takes you to the grid grid from where the total is being calculated.
Is there a way to take advantage of the quick filter to auto filter based on a particular column in the grid
NOTE, please when you navigate from the report menu there shouldnt be any automaticic filter, its only when the link from the dashboard page is clicked then the filter should apply.

Dashboard image
![image](https://github.com/serenity-is/Serenity/assets/50826372/90483106-c6d4-4c7c-99d9-4310f1b73115)

Auto filtering grid page

![image](https://github.com/serenity-is/Serenity/assets/50826372/e4159adf-ff74-489c-bf2b-cf6296d31fec)

**Comment by luongnv6** [comment]
pls see this demo
https://demo.serenity.is/Northwind/Product?cat=1

**Reply by rhemasystems** [reply]
Thank you, I have seen that but hw do I achieve that from a url,not from the sub menu

**Reply by luongnv6** [reply]
add this to your NavigationItem
```C#
[assembly: NavigationLink(3000, "Customer/Customer 1", url: "~/customer?id=1", permission: "Customer:View", icon: "fa-user", ItemClass = "customer-menu")]
```

**Reply by rhemasystems** [reply]
Thanks luongnv6,
Below is the link am clicking in a div to navigate to the desired page
![image](https://github.com/serenity-is/Serenity/assets/50826372/22be0754-d57e-46bc-8943-9e76737c0627)

How do I set the url to bring me to the below page with prefiltered column

![image](https://github.com/serenity-is/Serenity/assets/50826372/dd97bd10-e145-4b06-a440-d3787d13247d)

**Comment by luongnv6** [comment]
href="/Sales/Sales?cutomerId=1"

**Reply by rhemasystems** [reply]
Thanks, but I got this reply, there is no filter

![image](https://github.com/serenity-is/Serenity/assets/50826372/73d3ae8a-9634-4da2-94ba-bda347757f86)

**Reply by luongnv6** [reply, marked as answer]
pls follow this demo
https://github.com/serenity-is/common-features/blob/342c8110d92bf97e60b4e7cac06280b3c2cc5392/src/Serenity.Demo.Northwind/Modules/Product/ProductGrid.ts#L279

**Comment by rhemasystems** [comment]
Thank you loungnv6,
I combined your two answers and came up with a workable solution

![image](https://github.com/serenity-is/Serenity/assets/50826372/c92e1fb8-c9cf-47c4-88ee-0c8da563af16)

And used the void  createQuickFIlter()


       `protected createQuickFilters(): void {
            super.createQuickFilters();
            var url = decodeURI(window.location.href);
            var index = url.lastIndexOf('Sales?tenant=');
            var tenant = Q.toId(url.substring(index).replace('Sales?tenant=', ''));
            if (index > -1) {
                this.findQuickFilter(Serenity.StringEditor, fld.TenantName).value = tenant;
            }  
        }`

**Reply by luongnv6** [reply]
you can also get param like in demo:
```ts
protected getQuickFilters() {
        var filters = super.getQuickFilters();

        var q = parseQueryString();
        if (q["tenant"]) {
            var tenantNameFilter= tryFirst(filters , x => x.field == fld.TenantName);
            tenantNameFilter.init = e => {
                e.element.getWidget(Serenity.StringEditor).value = q["tenant"];
            };
        }
        return filters;
    }
```
